clouddriver-openstack can't create a server group without Telemetry Alarming service.

**Title**

clouddriver-openstack can't create a server group without Telemetry Alarming service.

**Cloud Provider**

OpenStack

**Environment**

I'm running Spinnaker using docker-compose on localhost(MacOSX)

```
% docker-compose images
 Container             Repository              Tag       Image Id        Size
---------------------------------------------------------------------------------
clouddriver   quay.io/spinnaker/clouddriver   master   80f763152c86   1.01e+03 MB
deck          quay.io/spinnaker/deck          master   457884327e46   910 MB
echo          quay.io/spinnaker/echo          master   06eff3905287   678 MB
fiat          quay.io/spinnaker/fiat          master   ab94efb56218   656 MB
front50       quay.io/spinnaker/front50       master   071216db7326   761 MB
gate          quay.io/spinnaker/gate          master   177deafa2447   675 MB
igor          quay.io/spinnaker/igor          master   8c3bf097f55d   657 MB
orca          quay.io/spinnaker/orca          master   3cc81bd8ecf2   678 MB
redis         redis                           latest   83744227b191   94.3 MB
rosco         quay.io/spinnaker/rosco         master   9fdb49e0224f   758 MB
```

**Feature Area**

Document

**Description**

I tried to create new server group, however get error message
(createServerGroup failed: Unable to process request).

**Steps to Reproduce**

* Deploy Spinnaker with clouddriver-openstack.
* Access http://localhost:9000
* Create Application
* Create LoadBalancer
* Create Server Group

**Additional Details**

Prerequisites[1] of Deploy OpenStack is written to need "Ceilometer"
[1]: https://www.spinnaker.io/setup/providers/openstack/ 

```
The Spinnaker Openstack driver is developed and tested against the Openstack Mitaka release. Due to the limitless ways to configure Openstack and its services, here is a list of API versions that are required to be enabled:

Keystone (Identity) v3
Compute v2
LBaaS v2
Networking v2
Orchestration (Heat)
Ceilometer
Glance v2
```
However, as I looked at the following log, I think clouddriver-openstack also needs Telemetry Alarming service (aodh).

```
clouddriver    | 2017-06-15 07:38:34.873 ERROR 1 --- [ool-12-thread-4] c.n.s.c.o.DefaultOrchestrationProcessor  : com.netflix.spinnaker.clouddriver.openstack.deploy.exception.OpenstackOperationException: createServerGroup failed: Unable to process request
clouddriver    | 	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
clouddriver    | 	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
clouddriver    | 	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
clouddriver    | 	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
clouddriver    | 	at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:83)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:105)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallConstructor(CallSiteArray.java:60)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:235)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:255)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.openstack.deploy.ops.servergroup.DeployOpenstackAtomicOperation.operate(DeployOpenstackAtomicOperation.groovy:187)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.openstack.deploy.ops.servergroup.DeployOpenstackAtomicOperation.operate(DeployOpenstackAtomicOperation.groovy)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation$operate.call(Unknown Source)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.orchestration.DefaultOrchestrationProcessor$_process_closure1$_closure2.doCall(DefaultOrchestrationProcessor.groovy:89)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.orchestration.DefaultOrchestrationProcessor$_process_closure1$_closure2.doCall(DefaultOrchestrationProcessor.groovy)
clouddriver    | 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
clouddriver    | 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
clouddriver    | 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
clouddriver    | 	at java.lang.reflect.Method.invoke(Method.java:498)
clouddriver    | 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
clouddriver    | 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
clouddriver    | 	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:294)
clouddriver    | 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1024)
clouddriver    | 	at groovy.lang.Closure.call(Closure.java:414)
clouddriver    | 	at groovy.lang.Closure.call(Closure.java:408)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.metrics.TimedCallable$ClosureWrapper.call(TimedCallable.groovy:55)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.metrics.TimedCallable.call(TimedCallable.groovy:82)
clouddriver    | 	at java_util_concurrent_Callable$call.call(Unknown Source)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:117)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.orchestration.DefaultOrchestrationProcessor$_process_closure1.doCall(DefaultOrchestrationProcessor.groovy:88)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.orchestration.DefaultOrchestrationProcessor$_process_closure1.doCall(DefaultOrchestrationProcessor.groovy)
clouddriver    | 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
clouddriver    | 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
clouddriver    | 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
clouddriver    | 	at java.lang.reflect.Method.invoke(Method.java:498)
clouddriver    | 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
clouddriver    | 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
clouddriver    | 	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:294)
clouddriver    | 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1024)
clouddriver    | 	at groovy.lang.Closure.call(Closure.java:414)
clouddriver    | 	at groovy.lang.Closure.call(Closure.java:408)
clouddriver    | 	at com.netflix.spinnaker.security.AuthenticatedRequest.lambda$propagate$1(AuthenticatedRequest.java:66)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.metrics.TimedCallable.call(TimedCallable.groovy:82)
clouddriver    | 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
clouddriver    | 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
clouddriver    | 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
clouddriver    | 	at java.lang.Thread.run(Thread.java:745)
clouddriver    | Caused by: com.netflix.spinnaker.clouddriver.openstack.deploy.exception.OpenstackProviderException: Unable to process request
clouddriver    | 	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
clouddriver    | 	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
clouddriver    | 	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
clouddriver    | 	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
clouddriver    | 	at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:83)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:105)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:255)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.openstack.client.OpenstackRequestHandler$Trait$Helper.handleRequest(OpenstackRequestHandler.groovy:40)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.openstack.client.OpenstackRequestHandler$Trait$Helper$handleRequest.call(Unknown Source)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.openstack.client.OpenstackRequestHandler$Trait$Helper$handleRequest.call(Unknown Source)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.openstack.client.OpenstackOrchestrationV1Provider.handleRequest(OpenstackOrchestrationV1Provider.groovy)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.openstack.client.OpenstackOrchestrationV1Provider$handleRequest.callCurrent(Unknown Source)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.openstack.client.OpenstackOrchestrationV1Provider.deploy(OpenstackOrchestrationV1Provider.groovy:40)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.openstack.client.OpenstackOrchestrationProvider$deploy$0.call(Unknown Source)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.openstack.client.OpenstackClientProvider.deploy(OpenstackClientProvider.groovy:33)
clouddriver    | 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
clouddriver    | 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
clouddriver    | 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
clouddriver    | 	at java.lang.reflect.Method.invoke(Method.java:498)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrap.invoke(PogoMetaMethodSite.java:190)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.call(PogoMetaMethodSite.java:71)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.openstack.deploy.ops.servergroup.DeployOpenstackAtomicOperation.operate(DeployOpenstackAtomicOperation.groovy:178)
clouddriver    | 	... 41 more
clouddriver    | Caused by: ClientResponseException{message=HEAT-E99001 Service aodh is not available for resource type OS::Ceilometer::Alarm, reason: aodh alarming endpoint is not in service catalog., status=400, status-code=BAD_REQUEST}
clouddriver    | 	at org.openstack4j.core.transport.HttpExceptionHandler.mapException(HttpExceptionHandler.java:38)
clouddriver    | 	at org.openstack4j.core.transport.HttpExceptionHandler.mapException(HttpExceptionHandler.java:23)
clouddriver    | 	at org.openstack4j.core.transport.HttpEntityHandler.handleLessThan500(HttpEntityHandler.java:101)
clouddriver    | 	at org.openstack4j.core.transport.HttpEntityHandler.handle(HttpEntityHandler.java:47)
clouddriver    | 	at org.openstack4j.core.transport.HttpEntityHandler.handle(HttpEntityHandler.java:24)
clouddriver    | 	at org.openstack4j.connectors.jersey2.HttpResponseImpl.getEntity(HttpResponseImpl.java:63)
clouddriver    | 	at org.openstack4j.openstack.internal.BaseOpenStackService$Invocation.execute(BaseOpenStackService.java:226)
clouddriver    | 	at org.openstack4j.openstack.internal.BaseOpenStackService$Invocation.execute(BaseOpenStackService.java:208)
clouddriver    | 	at org.openstack4j.openstack.heat.internal.StackServiceImpl.create(StackServiceImpl.java:31)
clouddriver    | 	at org.openstack4j.api.heat.StackService$create$0.call(Unknown Source)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.openstack.client.OpenstackOrchestrationV1Provider$_deploy_closure1.doCall(OpenstackOrchestrationV1Provider.groovy:51)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.openstack.client.OpenstackOrchestrationV1Provider$_deploy_closure1.doCall(OpenstackOrchestrationV1Provider.groovy)
clouddriver    | 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
clouddriver    | 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
clouddriver    | 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
clouddriver    | 	at java.lang.reflect.Method.invoke(Method.java:498)
clouddriver    | 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
clouddriver    | 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
clouddriver    | 	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:294)
clouddriver    | 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1024)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:42)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:57)
clouddriver    | 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:117)
clouddriver    | 	at com.netflix.spinnaker.clouddriver.openstack.client.OpenstackRequestHandler$Trait$Helper.handleRequest(OpenstackRequestHandler.groovy:34)
clouddriver    | 	... 58 more
```

--- a/setup/providers/openstack.md
+++ b/setup/providers/openstack.md
@@ -19,6 +19,7 @@ here is a list of API versions that are required to be enabled:
 * Networking v2
 * Orchestration (Heat)
 * Ceilometer
+* Telemetry Alarming (Aodh)
 * Glance v1
 
 You will need an account admin permissions for Spinnaker to use. You can download the [openrc](https://docs.openstack.org/user-guide/common/cli-set-environment-variables-using-openstack-rc.html) from your Horizon UI. To test your setup, use the the [OpenStack command line client](https://docs.openstack.org/developer/python-openstackclient/).

